### ISSUE TITLE
Add bucklescript

### ISSUE THREAD

**Issue by jchavarri** [fhelwanger/ocaml-on-windows#17]
One of the things @superherointj has figured out is that, in order for Merlin to work, `bucklescript` should be running on WSL as well.

The problem that @fhelwanger mentioned in [this comment](https://github.com/reasonml-editor/vscode-reasonml/issues/102#issuecomment-332624860):
> Something must have changed lately that is handling \ as an escape char

is due to Bucklescript being executed from Windows, while merlin being executed from Linux, and interpreting file paths differently.

@fhelwanger Would it be very hard to add BuckleScript to this project? I think that would help streamlining the installation on Windows + it would also guarantee that all binaries are "seeing" file paths etc from the same point of view, which would hopefully avoid issues. It makes sense to keep things consistent.

**Comment by fhelwanger** [comment]
I was thinking about that... the problem is that `bsb` is installed per project on `npm install`. So it would require running npm on WSL, right?

Everything else is installed globally 😕

I don't know how to make it work, because I don't want to mess with the npm version the user has.

**Comment by jchavarri** [comment]
The [official guide](https://reasonml.github.io/guide/javascript/quickstart) recommends to install it globally

![image](https://user-images.githubusercontent.com/220424/31349355-2cf1f5b4-ad23-11e7-8011-ef3717a7d3b5.png)

And by default, on the vscode extension, we're relying on `bsb` [being available globally](https://github.com/reasonml-editor/vscode-reasonml/blob/master/package.json#L73) too. So I think we should be good :) 

cc @chenglou for confirmation

**Comment by fhelwanger** [comment]
Nice, so I think we can add it.

**Comment by chenglou** [comment]
Yeah bs-platform is ideally installed globally for the tooling. You'd still model it as a devDependencies for contributors to `npm install` and contribute to the repos though.

We had the idea of having everything as a devDep, but it went overboard and asked folks to start their editor with the correct environment (the project's specific merlin, reason, bs versions, all modeled as devDeps). People didn't like doing that. Thus the few global binaries we ask people to install.

Isn't this still a problem? https://github.com/reasonml/reasonml.github.io/issues/195#issuecomment-335232981

Tldr yeah let's add it to the windows workflow.

**Comment by jchavarri** [comment]
From what @superherointj mentioned, the problem with PPX lines in `.merlin` including corrupted paths was due to `bsb` running on Windows _natively_ while `ocamlmerlin` running on WSL / Linux. This caused problems where a file path would include `\` instead of `/` which would lead to `merlin` interpreting that as escape chars and then everything breaking.

Installing `bsb` in WSL seemed to fix it, hence this issue.

I think the Reason "install on Windows" docs should include clearly that all the tooling needs to run on WSL, to avoid this kind of situations.

_Disclaimer: i have no Windows machine_ 😂  _this is just based on what I discussed with superhero._

**Comment by fhelwanger** [comment]
I tried to add bs-platform but it is failing to install on WSL.

Does anyone know what is causing this? I ran `sudo npm install -g bs-platform`

I'm using node 8.6 and npm 5.4.2.

<details>
> bs-platform@1.9.3 postinstall /usr/lib/node_modules/bs-platform
> node scripts/install.js

Working dir /usr/lib/node_modules/bs-platform
fs.js:773
  return binding.rename(pathModule._makeLong(oldPath),
                 ^

Error: EACCES: permission denied, rename '/usr/lib/node_modules/bs-platform/vendor/ninja-build/ninja.linux64' -> '/usr/lib/node_modules/bs-platform/bin/ninja.exe'
    at Object.fs.renameSync (fs.js:773:18)
    at Object.<anonymous> (/usr/lib/node_modules/bs-platform/scripts/install.js:74:8)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:607:3
npm ERR! code ELIFECYCLE
</details>

**Comment by chenglou** [comment]
Is this relevant? https://github.com/BuckleScript/bucklescript/issues/2051

**Comment by fhelwanger** [comment]
I created #18 to add bs-platform.

I also removed ocaml-windows, because it requires ocaml 4.04.0. Now it's installing 4.02.3.

~~If anyone wish to test it:~~

~~`npm install -g fhelwanger/ocaml-on-windows#bsb`~~
